stream-chat-android-state/src/main/java/io/getstream/chat/android/state/plugin/factory/StreamStatePluginFactory.kt

```diff
@@ -135,15 +135,15 @@ public class StreamStatePluginFactory(
         val stateRegistry = StateRegistry.create(
             scope.coroutineContext.job, scope, clientState.user, repositoryFacade, repositoryFacade.observeLatestUsers()
         )
-        val logic =
-            LogicRegistry.create(
-                stateRegistry,
-                globalState,
-                clientState,
-                config.userPresence,
-                repositoryFacade,
-                chatClient
-            )
+        val logic = LogicRegistry.create(
+            stateRegistry = stateRegistry,
+            globalState = globalState,
+            userPresence = config.userPresence,
+            repos = repositoryFacade,
+            client = chatClient,
+            clientState = clientState,
+            coroutineScope = scope,
```

Comment:
We currently don't cancel children of the scope anywhere, however this will be addressed here: https://github.com/GetStream/stream-chat-android/issues/3931